kong/tools/utils.lua

```diff
@@ -1468,4 +1468,82 @@ do
 end
 _M.time_ns = time_ns
 
+
+local sha256
+do
+  local digest = require "resty.openssl.digest"
+  local sha256_digest
+
+  function sha256(key)
+    local _, bin, err
+    if not sha256_digest then
+      sha256_digest, err = digest.new("sha256")
+      if err then
+        return nil, err
+      end
+    end
+
+    bin, err = sha256_digest:final(key)
+    if err then
+      return nil, err
+    end
+
+    _, err = sha256_digest:reset()
+    if err then
+      sha256_digest = nil
+    end
+
+    return bin
+  end
+end
+_M.sha256 = sha256
+
+
+local sha256_hex
```

Comment:
`sha256_hex` `sha256_base64` `sha256_base64url` have the almost same pattern.

I think that we can define a common function, then pass the encode function into it to reduce duplicated code.

Reply:
done, thanks